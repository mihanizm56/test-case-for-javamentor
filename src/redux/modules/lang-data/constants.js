export const REMOVE_COORDS = "REMOVE_COORDS";
export const CHOOSE_COORDS = "CHOOSE_COORDS";
export const SAVE_ALL_ROUTES = "SAVE_ALL_ROUTES";
export const MAKE_NEW_OFFER = "MAKE_NEW_OFFER";
export const SAVE_CHOOSEN_COORDS_ROUTES = "SAVE_CHOOSEN_COORDS_ROUTES";
