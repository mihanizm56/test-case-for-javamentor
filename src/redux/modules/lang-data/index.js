export * from './actions';
export * from './constants';
export * from './middleware';
export * from './selectors';
export { default } from './reducer';
