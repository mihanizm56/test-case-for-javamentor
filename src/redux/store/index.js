import { applyMiddleware, compose, createStore } from 'redux'
import createSagaMiddleware from 'redux-saga'
import rootSaga from './root-saga.js'
import { rootReducer } from './root-reducer'

export const createAppStore = () => {
    const sagaMiddleware = createSagaMiddleware()

    const store = createStore(
        rootReducer,
        compose(
            applyMiddleware(sagaMiddleware),
            window.__REDUX_DEVTOOLS_EXTENSION__
                ? window.__REDUX_DEVTOOLS_EXTENSION__()
                : f => f
        )
    )

    sagaMiddleware.run(rootSaga)
    return store
}
