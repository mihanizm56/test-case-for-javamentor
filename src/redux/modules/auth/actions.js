import { LOGIN_STATE_SUCCESS, LOGIN_STATE_FAILED, LOGOUT } from "./constants";

export const loginSuccessAction = () => ({ type: LOGIN_STATE_SUCCESS });

export const loginFailedAction = () => ({ type: LOGIN_STATE_FAILED });

export const logoutAction = () => ({ type: LOGOUT });
