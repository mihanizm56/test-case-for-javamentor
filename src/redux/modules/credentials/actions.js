import { INPUT_DATA_OF_CARD, CLEAR_CARD_DATA, OPEN_CARD_FORM, CLOSE_CARD_FORM } from "./constants";

export const saveDataOfCardAction = (cardName, expDate, cardNumber, cvv) => ({
	type: INPUT_DATA_OF_CARD,
	payload: {
		cardName,
		expDate,
		cardNumber,
		cvv,
	},
});

export const openFormCardAction = () => ({
	type: OPEN_CARD_FORM,
});

export const closeFormCardAction = () => ({
	type: CLOSE_CARD_FORM,
});

export const clearCardDataAction = () => ({
	type: CLEAR_CARD_DATA,
});
