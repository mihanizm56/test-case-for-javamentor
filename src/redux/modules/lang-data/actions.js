import { FETCH_LANG_DATA, PUT_LANG_DATA } from "./constants";

export const fetchLangData = type => ({ type: FETCH_LANG_DATA, payload: type });
export const putLangData = data => ({ type: PUT_LANG_DATA, payload: data });
