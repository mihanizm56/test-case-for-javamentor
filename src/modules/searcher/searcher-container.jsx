import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchLangData, getLangData } from "../../redux/modules/lang-data";
import { listOptions as options } from "./constants";

class WrappedContainer extends Component {
  state = {
    selectedField: null
  };

  handleSearchClick = () => {
    console.log("check handleSearchClick, should start ");
  };

  handleSearchChange = ({ target: { value } }) => {
    console.log("check handleSearchChange", value);
  };

  render = () => {
    const { listData, children } = this.props;

    return children({
      handleSearchClick: this.handleSearchClick,
      handleSearchChange: this.handleSearchChange,
      listData,
      options
    });
  };
}

export const SearcherContainer = connect(
  state => ({ getLangData: getLangData(state) }),
  { fetchLangData }
)(WrappedContainer);
