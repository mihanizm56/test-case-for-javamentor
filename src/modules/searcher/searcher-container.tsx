import React from "react";
import { connect } from "react-redux";
import { fetchLangData, getLangData } from "../../redux/modules/lang-data";
import { listOptions as options } from "./constants";
import { SearcherViewProps, handleSearchChangeParams } from "./types";
import { LangCardType } from "../../components/molecules";

type SearcherContainerPropsType = {
  fetchLangData: (value: string) => void;
  langData: Array<LangCardType>;
  children: (object: SearcherViewProps) => Array<React.ReactNode>;
};

type SearcherContainerStateType = {
  selectedValue: string;
};
class WrappedContainer extends React.Component<
  SearcherContainerPropsType,
  SearcherContainerStateType
> {
  state = {
    selectedValue: ""
  };

  handleSearchClick = () => this.props.fetchLangData(this.state.selectedValue);

  handleSearchChange = ({ target: { value } }: handleSearchChangeParams) =>
    this.setState({ selectedValue: value });

  render = () => {
    const { selectedValue } = this.state;
    const { langData, children } = this.props;
    const { handleSearchClick, handleSearchChange } = this;

    return children({
      handleSearchClick,
      handleSearchChange,
      selectedValue,
      langData,
      options
    });
  };
}

export const SearcherContainer: any = connect(
  getLangData,
  { fetchLangData }
)(WrappedContainer);
