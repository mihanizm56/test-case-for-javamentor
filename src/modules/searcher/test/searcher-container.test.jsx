import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import { SearcherContainer } from "../searcher-container";
import { createAppStore } from "../../../redux/store";
import { fetchLangData } from "../../../redux/modules/lang-data/actions";

jest.mock("../../../redux/modules/lang-data/actions", () => ({
  fetchLangData: jest.fn().mockReturnValue({ type: "test_action" })
}));

describe("test SearcherContainer", () => {
  let container;
  const store = createAppStore();

  beforeEach(() => {
    jest.clearAllMocks();
    container = shallow(
      <Provider store={store}>
        <SearcherContainer>
          {objectWithProps => <p className="test-prop" {...objectWithProps} />}
        </SearcherContainer>
      </Provider>
    );
  });

  describe("test setting props to a component", () => {
    it("test set handleSearchChange prop ", () => {
      expect(
        container
          .dive()
          .dive()
          .dive()
          .dive()
          .find(".test-prop")
          .prop("handleSearchChange")
      ).toBeInstanceOf(Function);
    });
    it("test set handleSearchChange prop ", () => {
      expect(
        container
          .dive()
          .dive()
          .dive()
          .dive()
          .find(".test-prop")
          .prop("handleSearchClick")
      ).toBeInstanceOf(Function);
    });
    it("test set handleSearchChange prop ", () => {
      expect(
        container
          .dive()
          .dive()
          .dive()
          .dive()
          .find(".test-prop")
          .prop("langData")
      ).toBeInstanceOf(Array);
    });
    it("test set handleSearchChange prop ", () => {
      expect(
        container
          .dive()
          .dive()
          .dive()
          .dive()
          .find(".test-prop")
          .prop("options")
      ).toBeInstanceOf(Array);
    });
    it("test set handleSearchChange prop ", () => {
      expect(
        container
          .dive()
          .dive()
          .dive()
          .dive()
          .find(".test-prop")
          .prop("selectedValue")
      ).toBe("");
    });
  });
  describe.only("test action contract", () => {
    it("shoult do fetchLangData action", () => {
      container
        .dive()
        .dive()
        .dive()
        .dive()
        .find(".test-prop")
        .prop("handleSearchClick")();

      expect(fetchLangData).toHaveBeenCalledTimes(1);
    });
  });
});
