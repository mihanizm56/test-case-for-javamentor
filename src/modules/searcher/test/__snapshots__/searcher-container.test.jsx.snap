// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SearcherContainer test setting props to a component test set handleSearchChange prop  1`] = `
<p
  className="test-prop"
  handleSearchChange={[Function]}
  handleSearchClick={[Function]}
  langData={Array []}
  options={
    Array [
      Object {
        "value": "Front-end",
      },
      Object {
        "value": "Back-end",
      },
      Object {
        "value": "Mobile",
      },
    ]
  }
  selectedValue=""
/>
`;
